linode/manager
# upcoming: [DI-22184] - Added Metric Criteria, Metric components

## Description 📝
- Added the MetricCriteria and Metric components 
- Added new Unit Tests for the added components and added more tests and checks for existing components to reflect the latest changes
- Changed the types and validations to reflect the latest API spec

## Changes  🔄
- Added the Metric Criteria component for the Criteria Section using FieldArray to add the Metric components dynamically
- Added Autocomplete and TextField inputs for Data Field, Aggregation Type, Operator and Threshold in Metric
- Added the relevant test cases for Unit Tests 
- Added and changed some types of  CreateAlertDefinitionPayload, Alert interfaces

## Target release date 🗓️

Please specify a release date (and environment, if applicable) to guarantee timely review of this PR. If exact date is not known, please approximate and update it as needed.

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/user-attachments/assets/8df51aa8-74bd-4680-9b11-3f82eb44455c) | ![image](https://github.com/user-attachments/assets/a5520d70-226d-45e0-ac97-5d8dd5933ce5)|

## How to test 🧪

### Prerequisites

(How to setup test environment)

- The tabs are controlled by a featureFlag called aclpAlerting, the flag has been currently disabled. For testing enable the definitions part of the aclpAlerting flag to be true.
- The API endpoints are not live yet, so use Legacy MSW Handlers for the mock responses

### Verification steps
 - In Monitor (once the prerequisites are followed) , Alerts tab should be visible next to Dashboards.
- Under that, Definition tab and a Create button should be visible.
- On clicking Create, the Form Page should be visible
- To choose from the Data Field drop down, selecting a Service is required
- To choose from the Aggregation Field drop down, selecting a Data Field is required.
- The Metric component can be added and removed dynamically with the Add Metric and Delete Icon buttons
- The Delete Icon button is only visible when there is more than one Metric component

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
